src/lib.rs

```diff
@@ -114,10 +120,22 @@ macro_rules! flate {
 pub fn decode(bytes: &[u8]) -> Vec<u8> {
     use std::io::{Cursor, Read};
 
-    let mut dec = deflate::Decoder::new(Cursor::new(bytes));
     let mut ret = Vec::new();
-    dec.read_to_end(&mut ret)
-        .expect("Compiled DEFLATE buffer was corrupted");
+
+    #[cfg(feature = "zstd")]
```

Comment:
Cargo features are additive. What if someone uses two libraries, each requesting a different feature?
I think the compression algorithm should be configured at call site instead of compilation flags. Features should only affect whether an algorithm is supported, not whether it is used

Reply:
Thank you for your feedback. Firstly, I agree with your opinion overall. This is good point, I were seeking to how this feature should be implemented. I choose the Cargo features because I believe there should not be the unnecessary crate in the dependency. This is more important if it becomes an C-binding crate like [`zstd-rs`](https://github.com/gyscos/zstd-rs). The approach is similar to the [`zip-rs`](https://github.com/zip-rs/zip/blob/3e88fe66c941d411cff5cf49778ba08c2ed93801/Cargo.toml#L36-L42).

Alternatively, we can implement a wrapper for `Decoder` and `Encoder`. I am not sure that way would satisfy you. Could you tell the specifics of what you're think ideal?

Reply:
Instead of Cargo features, we can do something like this to make the compression method selective for each files, however, always depending on the potentially-unwanted crates as a result, if acceptable:
```rust
flate!(pub static DATA: [u8] from "assets/009f.dat"); // default, DEFLATE
flate!(pub static DATA: [u8] from "assets/009f.dat" include_flate::Compression::Zstd); // Use Zstd for this file spcifically
flate!(pub static DATA: [u8] from "assets/009f.dat" include_flate::Compression::Deflate); // Explicitly use DEFLATE.
```

Reply:
Looks good to me, but perhaps we could omit the namespace as well since it's just a keyword read by the proc-macro.

Reply:
Sure thing, what about a some string?
```rust
flate!(pub static DATA: [u8] from "assets/009f.dat"); // default, DEFLATE
flate!(pub static DATA: [u8] from "assets/009f.dat" "zstd"); // Use Zstd for this file spcifically
flate!(pub static DATA: [u8] from "assets/009f.dat" "deflate"); // Explicitly use DEFLATE.
```

Reply:
or maybe just a keyword?

```rust
flate!(pub static DATA: [u8] from "assets/009f.dat"); // default, DEFLATE
flate!(pub static DATA: [u8] from "assets/009f.dat" with zstd); // Use Zstd for this file spcifically
flate!(pub static DATA: [u8] from "assets/009f.dat" with deflate); // Explicitly use DEFLATE.
```

```rust
($vis:vis static $ident:ident: $ty:ty from $file:literal $(with $algo:ident)?
```